From 9814c5890ecd3f1b044c234543010d8aa6f856b9 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 28 Jun 2020 11:14:10 +0000
Subject: [PATCH] GCC: add explicit namespace for FrameWidget

GCC is confused by multiple definitions in different namespaces
and fails like this:

../../third_party/blink/renderer/platform/widget/widget_base_client.h:153:24:
error: declaration of 'virtual blink::FrameWidget*
blink::WidgetBaseClient::FrameWidget()' changes meaning of 'FrameWidget'
[-fpermissive]
  153 |   virtual FrameWidget* FrameWidget() { return nullptr; }
      |                        ^~~~~~~~~~~
---
 third_party/blink/renderer/platform/widget/widget_base_client.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/platform/widget/widget_base_client.h b/third_party/blink/renderer/platform/widget/widget_base_client.h
index b4ff5ed..606deab 100644
--- a/third_party/blink/renderer/platform/widget/widget_base_client.h
+++ b/third_party/blink/renderer/platform/widget/widget_base_client.h
@@ -150,7 +150,7 @@ class WidgetBaseClient {
       mojo::PendingRemote<mojom::blink::WidgetInputHandlerHost> host) = 0;
 
   // The FrameWidget interface if this is a FrameWidget.
-  virtual FrameWidget* FrameWidget() { return nullptr; }
+  virtual blink::FrameWidget* FrameWidget() { return nullptr; }
 
   // Send the composition change to the browser.
   virtual void SendCompositionRangeChanged(
-- 
2.26.2

